### DISCUSSION TITLE
Is the `delete` fn description outdated?

### DISCUSSION THREAD

**Discussion by AspadaX** [diesel-rs/diesel#4140] [Q&A] [answered]
### Versions

- **Rust:** 1.8.0
- **Diesel:** 2.2.0
- **Database:** MySQL
- **Operating System** Ubuntu 22.04 Jammy

### Feature Flags

- **diesel:**

## Problem Description

The delete function in the official document seems to be outdated, as I eventually figured that this is the way makes it work: 
```
diesel::delete(
            schema::collection_lighteningdocs::dsl::collection_lighteningdocs
        )
            .filter(crate::schema::collection_lighteningdocs::dsl::collection_id.eq(collection_id))
            .execute(connection)?;
```
The one in the documentation as of Aug 1st, 2024:
```
diesel::delete(users.filter(id.eq(1))).execute(connection)?;
```

## Checklist

- [ ] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues) and the [discussion forum](https://github.com/diesel-rs/diesel/discussions) for similar possible closed issues.
 
- [ ] This issue can be reproduced on Rust's stable channel. (Your issue will be
  closed if this is not the case)
- [ ] This issue can be reproduced without requiring a third party crate

**Comment by weiznich** [comment, marked as answer]
I converted this to a discussion, as this is not a bug in diesel but a support question. Please fill future questions as discussions, as we use the issue tracker to track bugs. Any further question filled as issue will be handled as issue, so in that case closed as invalid.

To answer the actual question: Both variants are possible, the documentation is still accurate. We run the documention as part of our CI setup so the examples there are guaranteed to work.
